assert_release_quality

```diff
@@ -1,8 +1,20 @@
 #!/bin/sh
+#
+# Asserts that the module is ready for release with the expected quality
 
 # cf. https://docs.npmjs.com/cli/version
 new_version=$1
 
+# Checking that the npm-shrinkwrap.json file is up-to-date wrt to the
+# package.json file.
+if [ package.json -nt npm-shrinkwrap.json ]
+then
+    echo "File package.json is newer than file npm-shrinkwrap.json"
```

Comment:
I'm a bit confused about this. There are plenty of reasons to modify package.json that wouldn't require you to regenerate the npm-shrinkwrap.json file (which can take a couple minutes with rm-rf and reinstalling all node modules).

Unrelated, but we may want to add a note about npm2 vs npm3, since they use a very different file structure. /cc @jrgm